packages/now-build-utils/src/detect-routes.ts

```diff
@@ -292,11 +292,16 @@ export async function detectRoutes(
 ): Promise<RoutesResult> {
   const routesResult = await detectApiRoutes(files, builders);
   const publicBuilder = getPublicBuilder(builders);
+  const { defaultRoutes } = routesResult;
 
-  if (routesResult.defaultRoutes && publicBuilder) {
+  if (defaultRoutes && defaultRoutes.length > 0) {
+    defaultRoutes.unshift({ handle: 'miss' });
+  }
+
+  if (defaultRoutes && publicBuilder) {
     const directory = publicBuilder.src.replace('/**/*', '');
 
-    routesResult.defaultRoutes.push({
+    defaultRoutes.push({
```

Comment:
I'd prefer `routesResult.defaultRoutes` since updating a deconstructed reference wouldn't be as clear when the code gets more complex